src/test/java/net/mekomsolutions/senaite/integration/TestOrderFromOrderCreatorRouteITest.java

```diff
@@ -0,0 +1,73 @@
+package net.mekomsolutions.senaite.integration;
+
+import org.apache.camel.EndpointInject;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.apache.camel.builder.AdviceWithRouteBuilder;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.camel.model.ModelCamelContext;
+import org.apache.camel.model.RouteDefinition;
+import org.apache.camel.reifier.RouteReifier;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.eip.component.entity.Event;
+import org.springframework.beans.factory.annotation.Value;
+
+
+public class TestOrderFromOrderCreatorRouteITest extends BaseCamelContextSensitiveTest {  
+
+    @EndpointInject(value = "mock:labOrderEndpoint")
+    private MockEndpoint labOrderEndpoint;
+    
+    @EndpointInject(value = "mock:sqlEndpoint")
+    private MockEndpoint sqlEndpoint;
+    
+    @Value("${bahmni.test.orderType.uuid}")
+    private String bahmniTestOrderTypeUuid;
+    
+    @Before
+    public void setup() throws Exception {
+    	loadXmlDefinedRoute("test_order-from-order-creator-route.xml");  
+    	RouteDefinition routeDefinition = camelContext.adapt(ModelCamelContext.class).getRouteDefinitions().get(0);
+    	RouteReifier.adviceWith(routeDefinition, camelContext, new AdviceWithRouteBuilder() {
+    	    @Override
+    	    public void configure() throws Exception {
+
+    	    	weaveByToString("DynamicTo[{{openmrs.baseUrl}}/ws/rest/v1/order/${exchangeProperty.lab-order-uuid}]").replace().toD("mock:labOrderEndpoint");;
+    	    	weaveByToString("DynamicTo[sql:INSERT INTO test_order(order_id) VALUES (${exchangeProperty.lab-order-id})?dataSource=openmrsDataSource]").replace().toD("mock:sqlEndpoint");
+    	    }
+    	});
+    	
+    	labOrderEndpoint.whenAnyExchangeReceived(new Processor () {
+			@Override
+			public void process(Exchange exchange) throws Exception {
+				exchange.getIn().setBody("{\"orderType\": {\"uuid\": \"" + bahmniTestOrderTypeUuid + "\"}}");
+			}
+    		
+    	});
+    	labOrderEndpoint.expectedHeaderReceived(Exchange.HTTP_METHOD, "GET");
+    	labOrderEndpoint.expectedHeaderReceived("Authorization", "Basic YWRtaW46QWRtaW4xMjM=");
+    	labOrderEndpoint.expectedPropertyReceived("lab-order-uuid", "order-uuid");
+    	
+    	sqlEndpoint.expectedPropertyReceived("lab-order-id", "1");
+    }
+
+    @Test
+    public void shouldCreateTestOrderFromOrder() throws Exception {
+    	// setup
+    	Event event = new Event();
+    	event.setTableName("orders");
+    	event.setIdentifier("order-uuid");
+    	event.setOperation("c");
+    	event.setPrimaryKeyId("1");
+    	
+    	// replay
+    	producerTemplate.sendBody("direct:test_order-from-order-creator", event);
+    	
+    	// verify
+    	labOrderEndpoint.assertIsSatisfied();
```

Comment:
Doesn't camel test also allow you check the payload sent to the target endpoint?